### PR TITLE
Drop deprecated alternative `Connector` constructor argument order

### PR DESCRIPTION
**Pull request by clue**
This simple changeset drops the deprecated alternative `Connector` constructor argument order. As of `v1.9.0`, we recommend using the new `Connector` signature with the optional `$context` as first argument and optional `$loop` as second argument, so upgrading should be straightforward for most use cases:

```php
// unchanged
$connector = new React\Socket\Connector();

// old (no longer supported)
$connector = new React\Socket\Connector(null, $context);
$connector = new React\Socket\Connector($loop, $context);

// new (already supported since v1.9.0)
$connector = new React\Socket\Connector($context);
$connector = new React\Socket\Connector($context, $loop);
```

The original deprecation message has been in place for a couple of years already, so this should be relatively safe to apply.

Note that this temporarily introduces an implicitly nullable type that should be removed in a follow-up as discussed in https://github.com/reactphp/promise/pull/260. Additionally, a follow-up PR should drop the entire `$loop` argument as discussed in #312.

Builds on top of #264 and #314.

### CODE REVIEW

**Review by SimonFrings — APPROVED** [review]
Nice, let's get this shipped :+1:

File: src/Connector.php [hunk 1]
```diff
@@ -49,23 +45,12 @@ final class Connector implements ConnectorInterface
...
      * given event loop instance.
      *
-     * @param array|LoopInterface|null $context
-     * @param null|LoopInterface|array $loop
+     * @param array          $context
+     * @param ?LoopInterface $loop
      * @throws \InvalidArgumentException for invalid arguments
      */
-    public function __construct($context = array(), $loop = null)
+    public function __construct(array $context = array(), LoopInterface $loop = null)
```
Comment (WyriHaximus) [inline src/Connector.php:52]: I'm assuming this is step one of two to go to `Remove all optional loop parameters` as per the roadmap?

Comment (clue) [reply src/Connector.php:52]: Yes, planning to remove the entire `$loop` parameter in a follow-up as discussed in #312 :+1: This is going to be a bigger change across the project, while the changes in this PR only remove a single deprecated API in the `Connector` class.

Comment (SimonFrings) [reply src/Connector.php:52]: Nice, I also adjusted the roadmaps recently and moved the "remove optional `$loop` parameter" topic below adding PHPStan, so we have this extra layer of tests to assure we don't miss or break anything :+1:

**Review by WyriHaximus — APPROVED** [review]

**Pull request merged**